### DISCUSSION TITLE
Add ReactiveStreams and Kotlin Coroutines variants for DataLoader

### DISCUSSION THREAD

**Discussion by hantsy** [graphql-java/java-dataloader#119] [General]
Spring Graphql [added an variant for the reactive/webflux implementations](https://github.com/spring-projects/spring-graphql/blob/main/spring-graphql/src/main/java/org/springframework/graphql/execution/DefaultBatchLoaderRegistry.java). I hope the upstream project can consider ReactiveStreams API(`Publisher`) and Kotlin Couroutines(`suspend fun` or `Flow`) as alternatives of the existing return type( Java 8 `CompletionStage`) of a BatchLoader or MappedBatchLoader.

**Comment by bbakerman** [comment]
We dont support it because it would bring in a an extra dependencies.

However a CompletableFuture can be turned back into a Mono or Flow easily enough

**Reply by hantsy** [reply]
Java 9 add a Flow API to adapt ReactiveStreams, https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.html

**Reply by bbakerman** [reply]
Yup and we are Java 8 based

**Comment by koenpunt** [comment]
@bbakerman you closed this as resolved, but I don't believe it really is?

**Comment by hantsy** [comment]
Almost all java libs and frameworks I used in projects are moved to Java 11 and Java 17, I think this data-loader project should catch up the main stream.
